app/services/ReviewService.scala

```diff
@@ -46,11 +49,31 @@ class ReviewService @Inject() (repository: ApprovalRepository, reviewRepository:
     }
   }
 
-  def changeStatus(id: String, info: ApprovalProcessStatusChange): Future[RequestOutcome[Unit]] = {
-    repository.changeStatus(id, info) map {
-      case error @ Left(Errors(NotFoundError :: Nil)) => error
-      case Left(_) => Left(Errors(InternalServiceError))
-      case result => result
+  def changeStatus(id: String, currentStatus: String, info: ApprovalProcessStatusChange): Future[RequestOutcome[Unit]] = {
+
+    def getContentToUpdate: Future[Option[ApprovalProcess]] = {
+      repository.getById(id) map  {
+        case Right(process) => if (process.meta.status == currentStatus) Some(process) else None
```

Comment:
I'd implement a logger and write a warning here.

Reply:
added a message to say process not found